File: src/app/Models/document-request.ts
export interface DocumentRequest {
    id: string;
    employeeId: string;
    documentType: DocumentType;
    monthYear: string;
    status: DocumentStatus;
    filePath?: string;
    createdAt: string;
    username?: string;
}

export interface DocumentRequestDto {
    documentType: DocumentType;
    monthYear: string;
}

  // document-types.enum.ts
  export enum DocumentType {
    PAYSLIP = 'PAYSLIP',
    WORK_ATTESTATION = 'WORK_ATTESTATION',
    CERTIFICATE = 'CERTIFICATE'
}
  
export enum DocumentStatus {
    REQUESTED = 'REQUESTED',
    PENDING_APPROVAL = 'PENDING_APPROVAL',
    GENERATED = 'GENERATED',
    FAILED = 'FAILED',
    APPROVED = 'APPROVED'
}

File: Tunisys-Multi-Module-Project/document-management-service/src/main/java/org/example/documentmanagementservice/Model/DocumentRequest.java
package org.example.documentmanagementservice.Model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
public class DocumentRequest {
    @Id
    private UUID id = UUID.randomUUID();

    private String employeeId;

    @Enumerated(EnumType.STRING)
    private DocumentType documentType;

    private String monthYear;

    @Enumerated(EnumType.STRING)
    private DocumentStatus status;

    private String filePath;

    private LocalDateTime createdAt;

    @Transient
    private String username;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getEmployeeId() {
        return employeeId;
    }

    public void setEmployeeId(String employeeId) {
        this.employeeId = employeeId;
    }

    public DocumentType getDocumentType() {
        return documentType;
    }

    public void setDocumentType(DocumentType documentType) {
        this.documentType = documentType;
    }

    public String getMonthYear() {
        return monthYear;
    }

    public void setMonthYear(String monthYear) {
        this.monthYear = monthYear;
    }

    public DocumentStatus getStatus() {
        return status;
    }

    public void setStatus(DocumentStatus status) {
        this.status = status;
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String filePath) {
        this.filePath = filePath;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}